pkg/segment/query/processor/dataprocessor.go

```diff
@@ -357,10 +358,10 @@ func NewStreamstatsDP(options *structs.StreamStatsOptions) *DataProcessor {
 	}
 }
 
-func NewTimechartDP(options *structs.TimechartExpr) *DataProcessor {
+func NewTimechartDP(options *structs.QueryAggregators, queryTimeRange *dtypeutils.TimeRange, qid uint64) *DataProcessor {
 	return &DataProcessor{
```

Comment:
Yes, I need to pass the `qid` to the constructor, so that it does some pre-populating. But it is mainly for logging purposes.

I have pushed all these params as `timechartOptions` struct field and I am passing just options to `NewTimechartDP`